### DISCUSSION TITLE
3.0 -> 3.1 migration

### DISCUSSION THREAD

**Discussion by gufio** [saleor/saleor#9386] [General]
Hi, I have a few questions regarding migration Saleor (api/dashboard) from version 3.0 to 3.1.

First of all, we've been using Saleor since 2.11 version and since then we've done two fundamental migrations:

- 2.11 -> 3.0 b.24 (api) / 3.0 b.17 (dashboard)
- 3.0 b.24 (api) and 3.0 b.17 (dashboard) -> 3.0.0 official release

And now I try to migrate to official 3.1 release. I've noticed, that some commits are present on 3.0 branch (kinda fresh, like 2 months old) and they're missing on 3.1 branch and thats especially for dashboard. I wonder if these unported commits from 3.0 will land on 3.1 in future or 3.1 branch is more recent and thats totally fine? Maybe I've started migration too early? Was 3.1 branch developed
more loosely and separate from 3.0? And finally how would you do that migration from 3.0 to 3.1?

**Comment by jwm0** [comment]
Hi @gufio, we've taken steps to ensure all bugfixes/features added in 3.0 were also ported to 3.1. With hundreds of those commits ported between 3.0 and 3.1 it's possible we might have missed something. Could you pinpoint the missing commits you noticed yourself?

**Reply by gufio** [reply]
Hi @jwm0. Sadly, I didn't write down these commits, but here is one for dashboard: 173aee4d3112f921977d3dbc2bb76165c9451732.
